python/ray/experimental/state.py

```diff
@@ -435,27 +477,34 @@ def task_profiles(self, start=None, end=None, num=None):
                     if "worker_id" in event[3]:
                         task_info[task_id]["worker_id"] = event[3]["worker_id"]
                     if "function_name" in event[3]:
-                        task_info[task_id]["function_name"] = (
-                            event[3]["function_name"])
-                if heap_size > num:
+                        task_info[task_id]["function_name"] = \
+                            event[3]["function_name"]
+
+                if num_slice is not None and heap_size > num_slice:
                     min_task, task_id_hex = heapq.heappop(heap)
                     del task_info[task_id_hex]
                     heap_size -= 1
+
         return task_info
 
-    def dump_catapult_trace(self, path, start=None, end=None, num=None):
+    def dump_catapult_trace(self,
+                            path,
+                            task_info,
+                            breakdowns=False):
         """Dump task profiling information to a file.
 
-        This information can be viewed as a timeline of profiling information
-        by going to chrome://tracing in the chrome web browser and loading the
-        appropriate file.
+      This information can be viewed as a timeline of profiling information by
+      going to chrome://tracing in the chrome web browser and loading the
+      appropriate file.
+
+      Args:
+        path: The filepath to dump the profiling information to.
+        task_info: relevant task profile information for desired time window
+        breakdowns: boolean indicating whether to show tasks in components
+      """
+
+        # TO DO - convert info to deltas
```

Comment:
Does this TODO still need to be done?

If so, but it's going to be done in a different PR, then this should be formatted along the lines of `# TODO(alanamarzoev):`